p5/visualTests/sanity_testing.py

```diff
@@ -0,0 +1,39 @@
+import subprocess
+import os
+from subprocess import run, PIPE
+
+dirname = os.path.dirname(os.path.realpath(__file__))
+
+test_names_2d = [f'{dirname}/sanityTests/2DSanityTests/{name}' for name in
+                 os.listdir(os.path.join(dirname, 'sanityTests/2DSanityTests'))
+                 if name not in ['__init__.py', '__pycache__']]
+
+test_names_3d = [f'{dirname}/sanityTests/3DSanityTests/{name}' for name in
+                 os.listdir(os.path.join(dirname, 'sanityTests/3DSanityTests'))
+                 if name not in ['__init__.py', '__pycache__']]
+
+print('============================= Running 2D tests =============================')
+for name in test_names_2d:
+    print('Running', name)
+    try:
+        # Run each test for 2s then raise a timeout
+        process = run(['python', name], stderr=PIPE, encoding='utf-8', timeout=2)
+    except subprocess.TimeoutExpired:
+        pass
+    else:
```

Comment:
I tested this locally but wasn't able to get the program to error even if I tried to manually break a function in p5. I think this is because vispy is capturing the exception is internally. As to why there is also no output to stderr - I'm not sure.

This said, I found a way to disable this exception capturing in vispy: add `self.events.ignore_callback_errors = False` to Vispy2DRenderer/base.py:69. Then, you should be able to just raise any exception here that's not `subprocess.TimeoutExpired`.

Reply:
After adding some breaking changes, did you reinstalled the project again ? `pip install .`
The tests are referring to the installed p5 package instead of the current contents in project. Maybe that was the reason you could not get it to error. Can you confirm this ? 
To use the current project, without reinstalling again, we can export the project's path to PYTHONPATH locally. 
Thanks.

Reply:
Its weird, vispy's window doesn't terminates after raising an error. I tried `self.events.ignore_callback_errors = False`, but it is not showing any difference. It logs there is an error but doesn't exits with a non-zero exit code.
Also found #89 related to it.

Reply:
Yes, I confirm that I did `pip install .` after every change. I went back and tested `self.events.ignore_callback_errors = False` again and it seems that it is not the only `EmitterGroup` that may be intercepting exceptions - there is another one in the `timer`. Could you try adding `self.timer.events.ignore_callback_errors = False` to `VispySketch.__init__` as well?

Reply:
Works!!!
Thanks a ton @ziyaointl 
Any chances there might be other emitter groups, we have to disable callbacks ?

Reply:
You are welcome! A quick search indicates that the vispy `Visual` may also contain an `EmitterGroup`, but not sure if p5py uses that at all. https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/vispy/vispy%24+EmitterGroup&patternType=literal
I think it could be sufficient for us to turn more of this flag off if we encounter more unraised Exceptions later.

Reply:
Great! This also seems to be the solution for #89